include/mls/credential.h

```diff
@@ -38,6 +41,32 @@ struct BasicCredential
   static const CredentialType type;
 };
 
+///
+/// X509 Credential
+///
+
+// case x509:
+//     opaque cert_data<1..2^24-1>;
+struct X509Credential
+{
+
+  X509Credential() {}
+  explicit X509Credential(const std::vector<bytes>& chain_in);
+
+  std::vector<std::unique_ptr<X509Certificate>> chain;
```

Comment:
Let's just make this `std::vector<bytes>`, and not even pretend to interpret the contents.

The Credential will need to interpret the leaf, to recognize the signature algorithm and extract the public key.  But it should just do that with OpenSSL on creation and store the public key thereafter.